client/allocrunner/taskrunner/consul_hook.go

```diff
@@ -79,7 +79,8 @@ func (h *consulHook) Prestart(ctx context.Context, req *interfaces.TaskPrestartR
 			}
 
 			env := map[string]string{
-				"CONSUL_TOKEN": token.SecretID,
```

Comment:
note for reviewers: as I mention in the docs update below, we're leaving this old variable for backwards compatibility.